browser/search_engines/tor_window_search_engine_provider_service.cc

```diff
@@ -18,9 +18,9 @@ TorWindowSearchEngineProviderService(Profile* otr_profile)
     : SearchEngineProviderService(otr_profile) {
   DCHECK(otr_profile->IsTor());
 
-  auto provider_data = GetInitialSearchEngineProvider(otr_profile->GetPrefs());
-  alternative_search_engine_url_for_tor_ =
-      std::make_unique<TemplateURL>(*provider_data);
+  auto provider_data = TemplateURLDataFromPrepopulatedEngine(
```

Comment:
Is this using `search.brave.com` or `search.brave4u7jddbv7cyviptqjc7jusxh72uik7zt6adtckl5f4nwy2v72qd.onion`?

We should probably use the latter since it gives us a better ability to limit CAPTCHAs.

Reply:
Addressed by https://github.com/brave/brave-core/pull/11916/commits/42be824c9fa02b99834836953b735141ec030c86.
Only tor window will use onion brave search url.
To do this, `brave_search_tor` is newly defined and only tor window uses that engine config.
@bsclifton WDYT?

Reply:
@simonhong that change LGTM! 👍